pkgs/stdenv/generic/setup.sh

```diff
@@ -855,6 +855,39 @@ stripHash() {
 }
 
 
+propagatePropagatedDependencies() {
```

Comment:
Maybe something like `{persist,record,output}PropagatedDependencies` so that we are not propagating what is already propagated?

Reply:
changed to `recordPropagatedDependencies`